### PR TITLE
Lock dialog refreshes lock data when opened

### PR DESCRIPTION
**Pull request by maureensturgeon**
Teacher complaint: "I unlock the assessment for the whole class, then a student takes it, then decides to retake based on their score. If I click the “lock settings” button, the assessment shows as unlocked under Individual Student Control. I.e. the status in the “lock settings” window is not current. If I refresh the page before clicking the “lock settings” button, then the statuses in Individual Student Control reflects the correct status of locked for my students who have completed the assessment. So, in order to unlock an assessment for a student who completed the assessment, I first have to “lock” them, hit save, then unlock them and hit save again (or refresh the unit screen first, then unlock the student)."

TL;DR -> The lock dialog does not refresh when re-opened and the lock status has changed.

I've added a lock dialog refresh method which will re-query lock status when the dialog is opened. It may take a moment for the lock status to update since it depends on an ajax request and I didn't want to wait to render the dialog, but ultimately each time the dialog is opened, the lock statuses should be up-to-date.

Video shows lock dialog with editable assessment, student re-takes assessment, lock dialog re-opened and now assessment is locked:

https://user-images.githubusercontent.com/24235215/143093004-a971d6ae-a0ae-424b-8a69-16d3fc049722.mov

## Links

- jira ticket: [LP-1932](https://codedotorg.atlassian.net/browse/LP-1932)

## Testing story
Tested locally and added unit tests

## Follow-up work
Are we okay with a slight delay for updates to lock status or do we want a loading UI state?

Also - I created https://codedotorg.atlassian.net/browse/LP-2133 to add an unsubmit button to top of assessment to make it easier for the student to retake. We've had reports about issues with unsubmit and I found the current state annoying when I was walking through the steps to lock and unlock assessments while testing.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by Erin007 — APPROVED** [review]

**Review by jamescodeorg — COMMENTED** [review]
Thanks for including the video, that's really helpful!

I'm wondering if it will be difficult for the teacher (and us when troubleshooting) to determine if the refresh request has occurred.  We could add a spinner to indicate that a request was pending, but if were going to do _that_, then maybe we should also just always load the lock state asynchronously when the dialog is opened (and not have two separate code paths, one for initial loading and one for refreshing)?

In any case, I think this is a good incremental improvement that we can move ahead with but it kind of feels like an intermediate state that we'd want to improve again in the near future.
